lib/nugget/statsd.rb

```diff
@@ -9,24 +9,42 @@ def self.stats
 
     def self.send_metrics(name, result, response)
       statsd = Nugget::NStatsd.stats
-      send_test_result(statsd, name, result)
-      send_test_timings(statsd, name, response)
+      send_test_result(statsd, name, result, response)
+      send_test_timings(statsd, name, result, response)
     end
 
-    def self.send_test_result(statsd, name, result)
+    private
 
-      if result == "FAIL"
-        statsd.gauge("#{name}.failures.count", 1)
-        Nugget::Log.debug("Sending the following to statsd: #{name}_failure_count: 1")
-      else
-        statsd.gauge("#{name}.failures.count", 0)
-        Nugget::Log.debug("Sending the following to statsd: #{name}_failure_count: 0")
-      end
+    def self.send_test_result(statsd, name, result, response)
+      failure = (result == "FAIL")
+      dns_failure = failure && response.is_a?(Hash) && response[:return_code] == :couldnt_resolve_host
+      tcp_failure = failure && response.is_a?(Hash) && (response[:return_code] == :couldnt_connect || response[:return_code] == :operation_timedout)
+      tls_failure = failure && response.is_a?(Hash) && (response[:return_code] == :ssl_connect_error || response[:return_code] == :ssl_cacert)
+
+      # If we failed but for a non distinct protocol reason, it must be http
```

Comment:
Oops this isn’t true for `tcp` checks, only for those that _attempt_ an `http` check